btk/survey.py

```diff
@@ -0,0 +1,243 @@
+    assert band in bands
+    return survey.filters[bands.index(band)]
+
+
+def get_flux(ab_magnitude, filt, survey):
+    """Convert source magnitude to flux.
+    The calculation includes the effects of atmospheric extinction.
+    Credit: WeakLensingDeblending (https://github.com/LSSTDESC/WeakLensingDeblending)
+
+    Args:
+        ab_magnitude(float): AB magnitude of source.
+    Returns:
+        float: Flux in detected electrons.
+    """
+    zeropoint_airmass = 1.0
+    if survey.name == "DES":
+        zeropoint_airmass = 1.3
+    if survey.name == "LSST" or survey.name == "HSC":
+        zeropoint_airmass = 1.2
+    if survey.name == "Euclid":
+        zeropoint_airmass = 1.0
+
+    mag = ab_magnitude + filt.extinction * (filt.airmass - zeropoint_airmass)
+    return filt.exposure_time * filt.zero_point * 10 ** (-0.4 * (mag - 24))
+
+
+def get_mean_sky_level(survey, filt):
+    return get_flux(filt.sky_brightness, filt, survey) * survey.pixel_scale ** 2
+
+
+def make_wcs(pixel_scale, shape, center_pix=None, center_sky=None, projection="TAN"):
+    """Creates WCS for an image.
+    Args:
+        pixel_scale (float): pixel size in arcseconds
+        shape (tuple): shape of the image in pixels.
+        center_pix: tuple representing the center of the image in pixels
+        center_sky: tuple representing the center of the image in sky coordinates
+                     (RA,DEC) in arcseconds.
+        projection(str): projection type, default to TAN. A list of available
+                            types can be found in astropy.wcs documentation
+    Returns:
+        wcs: WCS
+    """
+    if center_pix is None:
+        center_pix = [(s + 1) / 2 for s in shape]
+    if center_sky is None:
+        center_sky = [0 for _ in range(2)]
+    w = WCS.WCS(naxis=2)
+    w.wcs.ctype = ["RA---" + projection, "DEC--" + projection]
+    w.wcs.crpix = center_pix
+    w.wcs.cdelt = [pixel_scale / 3600 for _ in range(2)]
+    w.wcs.crval = [c / 3600 for c in center_sky]
+    w.array_shape = shape
+    return w
+
+
+def get_moffat_psf(survey, filt, psf_stamp_size):
+    """Generates a psf as a Galsim object using the survey information"""
+    assert psf_stamp_size % 2 == 1
+
+    def psf_function(r):
+        return galsim.Moffat(2, r)
+
+    psf_obj = psf_function(filt.psf_scale).withFlux(1.0)
+    psf = psf_obj.drawImage(
+        nx=psf_stamp_size,
+        ny=psf_stamp_size,
+        method="no_pixel",
+        use_true_center=True,
+        scale=survey.pixel_scale,
+    ).array
+
+    # Make sure PSF vanishes on the edges of a patch that
+    # has the shape of the initial psf
+    psf = psf - psf[1, int(psf_stamp_size / 2)] * 2
+    psf[psf < 0] = 0
+    psf /= np.sum(psf)
+
+    # Generating an unintegrated galsim psf for the convolution
+    psf_obj = galsim.InterpolatedImage(
+        galsim.Image(psf), scale=survey.pixel_scale
+    ).withFlux(1.0)
+
+    return psf_obj
+
+
+def get_psf(survey):
```

Comment:
In line with my comment above, I find this particular function a bit confusing for a user who knows what kind of psf model he wants, as you need to input mirror diameter and other parameters you may not want (also, do you plan to implant those in the Filters ?). Maybe a solution would be to directly provide the PSF in the surveys : if users want to get a custom PSF, they just have to create a custom Survey which is not too hard. I was thinking the PSF could be inputted either directly as a galsim object or a filename. This would give more control to the user as to which PSF he wants to use. We may still reuse this function for default values of the PSF in our surveys.

Reply:
Additionally, it is not natural to expect a list of PSFs from a `get_psf` method. Especially when no one knows the order which they are returned within the list.

Either it should be renamed, or a filter name could be provided in the method call to return the PSF for that particular filter (makes more sense)

Reply:
I agree with you @thuiop , to avoid this PR getting too big I propose in the next PR we improve the psf model generation (@HironaoMiyatake and/or @mpaillassa  can help). Can you create an issue with your ideas? Thanks!

For now I will only add the bare-minimum to get a consistent PSF model up and running.